src/Oxpecker/ModelBinder.fs

```diff
@@ -14,299 +12,259 @@ open Microsoft.Extensions.Primitives
-        else
-            None
-
-    and getValueForArrayOfGenericType
-        (culture: CultureInfo)
-        (data: IDictionary<string, StringValues>)
-        (prop: PropertyInfo)
-        : obj option =
-
-        if prop.PropertyType.IsArray then
-            let regex = prop.Name |> Regex.Escape |> sprintf @"%s\[(\d+)\]\.(\w+)" |> Regex
-
-            let innerType = prop.PropertyType.GetElementType()
-
-            let arrOfValues =
-                data
-                |> Seq.filter(fun item -> regex.IsMatch item.Key)
-                |> Seq.map(fun item ->
-                    let matchedData = regex.Match item.Key
-                    let index = matchedData.Groups[1].Value
-                    let key = matchedData.Groups[2].Value
-                    let value = item.Value
-                    index, key, value)
-                |> Seq.groupBy(fun (index, _, _) -> index |> int)
-                |> Seq.sortBy fst
-                |> Seq.choose(fun (index, values) ->
-                    let dictData =
-                        values
-                        |> Seq.fold
-                            (fun (state: Map<string, StringValues>) (_, key, value) -> state.Add(key, value))
-                            Map.empty
-
-                    let model = Activator.CreateInstance(innerType)
-                    let res = parseModel model culture dictData
-
-                    match res with
-                    | Ok o -> Some(index, o)
-                    | Error _ -> None)
-                |> Seq.toArray
-
-            let arrayOfObjects =
-                if (arrOfValues |> Array.length > 0) then
-                    let arraySize = (arrOfValues |> Array.last |> fst) + 1
-                    let arrayOfObjects = Array.CreateInstance(innerType, arraySize)
-
-                    arrOfValues
-                    |> Array.iter(fun (index, item) -> arrayOfObjects.SetValue(item, index))
-
-                    arrayOfObjects
-                else
-                    Array.CreateInstance(innerType, 0)
+                        None)
+            |> dict
+
+        if matchedData.Count > 0 then Some matchedData else None
+
+    type private FieldSetter<'T> = delegate of CultureInfo * IDictionary<string, StringValues> * 'T -> unit
+
+    type private Parser<'T> = CultureInfo -> IDictionary<string, StringValues> -> 'T
+
+    let rec private mkParser<'T> () : Parser<'T> =
+        match cache.TryFind() with
+        | Some x -> x
+        | None ->
+            use ctx = cache.CreateGenerationContext()
+            mkParserCached<'T> ctx
+
+    and private mkParserCached<'T> (ctx: TypeGenerationContext) : Parser<'T> =
+        match ctx.InitOrGetCachedValue<Parser<'T>>(fun cell culture data -> cell.Value culture data) with
+        | Cached(value = v) -> v
+        | NotCached t ->
+            let v = mkParserAux<'T> ctx
+            ctx.Commit t v
+
+    and private mkParserAux<'T> (ctx: TypeGenerationContext) : Parser<'T> =
```

Comment:
And can you please call is `makeParserInner`, it's according to my codestyle in this repository